cli/src/commands/split.rs

```diff
@@ -226,6 +303,111 @@ pub(crate) fn cmd_split(
         commit_builder.write(tx.repo_mut())?
     };
 
+    if use_move_flags {
+        new(
+            ui,
+            tx,
+            target,
+            first_commit,
+            second_commit,
+            new_parent_ids,
+            new_child_ids,
+        )
+    } else {
+        legacy(ui, tx, target, first_commit, second_commit, parallel)
+    }
+}
+
+fn new(
+    ui: &mut Ui,
+    mut tx: WorkspaceCommandTransaction,
+    target: CommitWithSelection,
+    mut first_commit: Commit,
+    mut second_commit: Commit,
+    new_parent_ids: Vec<CommitId>,
+    new_child_ids: Vec<CommitId>,
+) -> Result<(), CommandError> {
+    let mut rewritten_commits: HashMap<CommitId, CommitId> = HashMap::new();
+    tx.repo_mut()
+        .set_rewritten_commit(target.commit.id().clone(), second_commit.id().clone());
+    rewritten_commits.insert(target.commit.id().clone(), second_commit.id().clone());
+    tx.repo_mut()
+        .transform_descendants(vec![target.commit.id().clone()], |rewriter| {
+            let old_commit_id = rewriter.old_commit().id().clone();
+            let new_commit = rewriter.rebase()?.write()?;
+            rewritten_commits.insert(old_commit_id, new_commit.id().clone());
+            Ok(())
+        })?;
+
+    let new_parent_ids: Vec<_> = new_parent_ids
+        .iter()
+        .map(|commit_id| rewritten_commits.get(commit_id).unwrap_or(commit_id))
+        .cloned()
+        .collect();
+    let new_child_ids: Vec<_> = new_child_ids
+        .iter()
+        .map(|commit_id| rewritten_commits.get(commit_id).unwrap_or(commit_id))
+        .cloned()
+        .collect();
+    let stats = move_commits(
+        tx.repo_mut(),
+        &MoveCommitsLocation {
+            new_parent_ids,
+            new_child_ids,
+            target: MoveCommitsTarget::Commits(vec![first_commit.id().clone()]),
+        },
+        &RebaseOptions {
+            empty: EmptyBehaviour::Keep,
+            rewrite_refs: RewriteRefsOptions {
+                delete_abandoned_bookmarks: false,
+            },
+            simplify_ancestor_merge: false,
+        },
+    )?;
+
+    let mut num_new_rebased = 1;
+    if let Some(RebasedCommit::Rewritten(commit)) = stats.rebased_commits.get(first_commit.id()) {
+        first_commit = commit.clone();
+        num_new_rebased += 1;
+    }
+    if let Some(RebasedCommit::Rewritten(commit)) = stats.rebased_commits.get(second_commit.id()) {
+        second_commit = commit.clone();
+    }
+
+    if let Some(mut formatter) = ui.status_formatter() {
+        let mut num_rebased_descendants = rewritten_commits.len() + stats.rebased_commits.len();
+        // don't count the commit generated by the split in the rebased commits
+        num_rebased_descendants -= num_new_rebased;
+        for (_, ref rewritten) in rewritten_commits {
```

Comment:
nit: `rewritten_commits.values()`

It's also good to exclude duplicated entries upfront by `.iter().filter(..).count()`.